h/static/scripts/annotation-ui.js

```diff
@@ -77,15 +77,24 @@ var types = {
   SET_SORT_KEY: 'SET_SORT_KEY',
 };
 
+/**
+ * Return a copy of `current` with all matching annotations in `annotations`
+ * removed.
+ */
 function excludeAnnotations(current, annotations) {
-  var idsAndTags = annotations.reduce(function (map, annot) {
-    var id = annot.id || annot.$$tag;
-    map[id] = true;
-    return map;
-  }, {});
+  var idsAndTags = {};
+  annotations.forEach(function (annot) {
+    if (annot.id) {
+      idsAndTags[annot.id] = true;
+    }
+    if (annot.$$tag) {
+      idsAndTags[annot.$$tag] = true;
+    }
+  });
   return current.filter(function (annot) {
-    var id = annot.id || annot.$$tag;
-    return !idsAndTags[id];
+    var shouldRemove = (annot.id && idsAndTags[annot.id]) ||
```

Comment:
I know that it's not likely to happen, but there's a potential problem here if angular generates a tag that looks like an annotation ID.

It seems that it would be just as fast and just as clear to have an `ids` set and a `tags` set, rather than one joint `idsAndTags` set, and it would completely avoid this possibility.